app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectViewModel.kt

```diff
@@ -152,27 +157,54 @@ class WalletConnectViewModel(application: Application) : AndroidViewModel(applic
 
     fun loadTransactionFee() {
         binder?.getSessionRequestParams()?.parsePayload()?.let { payload ->
-            proxyRepository.getTransferCost(type = "update",
-                amount = payload.amount.toBigInteger(),
-                sender = walletConnectData.account!!.address,
-                contractIndex = payload.address.index,
-                contractSubindex = payload.address.subIndex,
-                receiveName = payload.receiveName,
-                parameter = payload.message,
-                success = {
-                    walletConnectData.energy = it.energy
-                    walletConnectData.cost = it.cost.toBigInteger()
-                    transactionFee.postValue(walletConnectData.cost)
-                },
-                failure = {
-                    handleBackendError(it)
+            when (payload) {
+                is Payload.ComplexPayload -> {
+                    proxyRepository.getTransferCost(type = "update",
+                        amount = payload.amount.toBigInteger(),
+                        sender = walletConnectData.account!!.address,
+                        contractIndex = payload.address.index,
+                        contractSubindex = payload.address.subIndex,
+                        receiveName = payload.receiveName,
+                        parameter = payload.message,
+                        success = {
+                            walletConnectData.energy = it.energy
+                            walletConnectData.cost = it.cost.toBigInteger()
+                            transactionFee.postValue(walletConnectData.cost)
+                        },
+                        failure = {
+                            handleBackendError(it)
+                        }
+                    )
                 }
-            )
+
+                is Payload.SimplePayload -> {
+                    proxyRepository.getTransferCost(
+                        type = ProxyRepository.SIMPLE_TRANSFER,
+                        memoSize = null,
+                        success = {
+                            walletConnectData.energy = it.energy
+                            walletConnectData.cost = it.cost.toBigInteger()
+                            transactionFee.postValue(walletConnectData.cost)
+                        },
+                        failure = {
+                            handleBackendError(it)
+                        }
+                    )
+                }
+            }
         }
     }
 
     fun hasEnoughFunds(): Boolean {
-        val amount = binder?.getSessionRequestParams()?.parsePayload()?.amount?.toBigInteger()
+        val payload = binder?.getSessionRequestParams()?.parsePayload()
+        val amount =
+            when (payload) {
+                is Payload.ComplexPayload -> payload.amount.toBigInteger()
+
+                is Payload.SimplePayload -> payload.amount.toBigInteger()
+
+                else -> BigInteger.ZERO
```

Comment:
Wouldn’t it be better to fall back to an error here?

Reply:
no need for that. i need to supply an excessive when here to have all the cases covered. we should never be in that situation, another PR adds event/chain checks to avoid this this case.